Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationController.java

```diff
@@ -0,0 +1,88 @@
+
+import edu.asu.spring.quadriga.domain.impl.networks.Transformation;
+import edu.asu.spring.quadriga.domain.network.tranform.ITransformation;
+import edu.asu.spring.quadriga.exceptions.FileStorageException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaGeneratorException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.network.transform.impl.Generator;
+import edu.asu.spring.quadriga.service.network.transform.impl.MatchGraphs;
+import edu.asu.spring.quadriga.service.network.transform.impl.TransformNode;
+import edu.asu.spring.quadriga.service.transformation.ITransformationManager;
+
+@Controller
+public class TransformationController {
+
+    private static final Logger logger = LoggerFactory.getLogger(TransformationController.class);
+
+    @Autowired
+    private MatchGraphs matchGraphs;
+
+    @Autowired
+    private ITransformationManager transformationManager;
+
+    @Autowired
+    private Generator generator;
+
+    /**
+     * Gets transformation-request,networks from the User and returns
+     * transformed graph Extensions: Obtain Mapping and Pattern files from the
+     * user.
+     * 
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws FileStorageException
+     */
+
+    @RequestMapping(value = "checks/transformation", method = RequestMethod.POST)
+    public String transformTest(@RequestParam("project") String project,
+            @RequestParam("transformation") String transformationNameAndID,
+            @RequestParam("sanitized_network_id") String networkID, Model model)
+            throws QuadrigaStorageException, IOException, QuadrigaGeneratorException, FileStorageException {
+
+        String[] transformationAndDirectory = transformationNameAndID.split(",");
+        String directoryName = null;
+
+        if (transformationAndDirectory.length > 1)
+            directoryName = transformationAndDirectory[1];
+
+        ArrayList<String> fileNames = transformationManager.retrieveTransformationFilePaths(directoryName);
+        String absolutePatternFilePath = fileNames.get(0);
+        String absoluteMappingFilePath = fileNames.get(1);
+
+        ITransformation transform = new Transformation();
+        transform.setPatternFilePath(absolutePatternFilePath);
+        transform.setTransformationFilePath(absoluteMappingFilePath);
+        List<ITransformation> transformations = new ArrayList<ITransformation>();
+        transformations.add(transform);
+        List<String> networkIdList = Arrays.asList(networkID.split("\\s*,\\s*"));
+        List<String> networkIds = new ArrayList<String>();
+        for (int i = 1; i < networkIdList.size(); i++) {
+            networkIds.add(networkIdList.get(i));
+        }
+
+        List<List<TransformNode>> results = matchGraphs.matchGraphs(transformations, networkIds);
+        logger.info("Check results for transformations: " + results);
```

Comment:
most of the logic in this controller should be in its own service classes that simply gets the ids of the transformations that should be run and the ids of the networks to run the transformations on and then runs the transformations.